smart_open/http.py

```diff
@@ -227,6 +240,8 @@ def __init__(self, url, mode='r', buffer_size=DEFAULT_BUFFER_SIZE,
         """
         self.url = url
 
+        self.session = session
```

Comment:
Like this, you don't need the new if statement below, as you can always call upon `self.session`:
```suggestion
        self.session = session or requests
```